File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n<=0) return 0;
          if( n==1)
               return 1; 
          int prev=1;
          int curr=1;

          for(int i=2;i<n+1;i++){
            int temp=curr;
            curr=prev+curr;
            prev=temp;
          }
          return curr;
    }
}

File: 70-climbing-stairs/climbing-stairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        memo={}
        return self.stairs(n,memo)
    def stairs(self,n,memo):
        if n==0 or n==1: return 1
        if n not in memo:
            memo[n]=self.stairs(n-1,memo)+self.stairs(n-2,memo)
        return memo[n]    
        